src/components/Community.js

```diff
@@ -1,16 +1,125 @@
+      personId && _.find(this.state.communityMembers, ['id', personId]);
+
+    if (featuredPerson) {
+      this.setState({ ...this.state, featuredPerson: featuredPerson });
+    } else {
+      // rotate who's featured every 10 seconds
+      this.intervalId = setInterval(() => {
+        let currIndex = _.findIndex(this.state.communityMembers, [
+          'id',
+          this.state.featuredPerson.id
+        ]);
+        let nextIndex =
+          currIndex + 1 === this.state.communityMembers.length
+            ? 0
+            : currIndex + 1;
+        this.setState({
+          ...this.state,
+          featuredPerson: this.state.communityMembers[nextIndex]
+        });
+      }, 10 * 1000);
+    }
+  }
+
+  componentDidMount() {
+    // scroll to community section if on a community URL
+    this.parsePersonIdFromURL() && this.scrollToCommunity();
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.intervalId);
+  }
+
+  render() {
+    return (
+      <section id="community">
+        <h2>Community</h2>
+        <div className="flex-container">
+          <div className="carousel-container">
+            <img
+              src={this.state.featuredPerson['image']}
+              alt={this.state.featuredPerson['name']}
+              className="responsive center"
+            />
+            <div className="bio">
+              {this.componentizedBio(this.state.featuredPerson)}
+            </div>
+          </div>
+          <div className="images-container">
+            {this.state.communityMembers.map(person => {
+              return (
+                <a
+                  key={person.id}
+                  href={`community/${person.id}`}
+                  onClick={e => this.handleImageClick(e, person)}
+                  className="image-overlay"
+                >
+                  <div className="overlay-text">{person['name']}</div>
+                  <img
+                    src={person['image']}
+                    alt={person['name']}
+                    className="community-image"
+                  />
+                </a>
+              );
+            })}
+          </div>
+        </div>
+      </section>
+    );
+  }
+
+  handleImageClick = (e, person) => {
+    e.preventDefault();
+    this.setState({ ...this.state, featuredPerson: person });
+    window.history.pushState(null, '', `/community/${person.id}`);
+    this.scrollToCommunity();
+    clearInterval(this.intervalId);
+  };
+
+  parsePersonIdFromURL = () => {
+    // parse pathname and return personId if on a community URL
+    let pathName = window.location.pathname.split('/');
+    return pathName[1] === 'community' ? pathName[2] : null;
+  };
+
+  scrollToCommunity() {
+    // wait until document is loaded before scrolling
+    if (
+      window.document &&
+      typeof window.document.createElement === 'function'
+    ) {
+      window.document.getElementById('community').scrollIntoView(true);
+    }
+  }
+
+  componentizedBio = person => {
```

Comment:
Love me some functional components! A pattern I often use for this (where the functional component is part of a class) is to name the method `render${COMPONENT_NAME}` – so this could become `renderBio` or `renderPerson`. Not requesting changes here, for the record, just laying out what's in my brain :)

Reply:
I like it -- will make that change!